source/DataAccess/Dummy/DummyReader.ts

```diff
@@ -3,7 +3,7 @@ import { inject, injectable } from 'inversify';
 import TYPES from '../../../ioc.types';
 import { Logger } from 'winston';
 
-import IDataReader from '../IDataReader';
+import IDataReader, { IQueryResult } from '../IDataReader';
```

Comment:
Can't these both be named imports (e.g. { IDataReader, IQueryResult })

Reply:
fwiw, I believe I've deleted DummyReader in the other branch and replaced with a ts-sinon stubInterface<IDataReader>

Reply:
I guess it was DummyWriter, but you might be able to use the same strategy here for less manual mock code